project_mis_report/models/project_project.py

```diff
@@ -0,0 +1,67 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, _
+from odoo.exceptions import ValidationError
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
+
+
+class ProjectProject(models.Model):
+    _inherit = "project.project"
+
+    mis_report_template_id = fields.Many2one('mis.report', string="Mis Report Template")
+    mis_report_instance_id = fields.Many2one('mis.report.instance', string="Mis Report")
+    is_group_account_manager = fields.Boolean(compute='_compute_is_group_account_manager')
+    is_group_project_manager = fields.Boolean(compute='_compute_is_group_project_manager')
+
+    def _compute_is_group_project_manager(self):
+        for record in self:
+            record.is_group_project_manager = True if record.user_has_groups(
+                'project.group_project_manager') else False
+
+    def _compute_is_group_account_manager(self):
+        for record in self:
+            record.is_group_account_manager = True if record.user_has_groups(
+                'account.group_account_manager') else False
+
+    def generate_monthly_dates(self):
+        result = []
+        current_date = self.date_start.replace(day=1)
+        end_date = self.date.replace(day=1)
+        end_date = end_date.replace(month=self.date.month + 1 if self.date.month < 12 else 1, year=self.date.year + 1 if self.date.month == 12 else self.date.year)

```

Comment:
Better try this:
```suggestion
        end_date = end_date + relativedelta(months=1)

```
as you actually do later, don't you?